udp/discover.go

```diff
@@ -40,14 +28,14 @@ func (s *Server) Discover(ctx context.Context, address, path string, receiverFun
 // DiscoveryRequest sends request to multicast/unicast address and wait for responses until request timeouts or server shutdown.
 // For unicast there is a difference against the Dial. The Dial is connection-oriented and it means that, if you send a request to an address, the peer must send the response from the same
 // address where was request sent. For Discover it allows the client to send a response from another address where was request send.
-func (s *Server) DiscoveryRequest(req *pool.Message, address string, receiverFunc func(cc *client.ClientConn, resp *pool.Message), opts ...MulticastOption) error {
+func (s *Server) DiscoveryRequest(req *pool.Message, address string, receiverFunc func(cc *client.ClientConn, resp *pool.Message), opts ...generic.MulticastOption) error {
```

Comment:
Please could you extend the test at https://github.com/plgd-dev/go-coap/blob/f28bb10a28501f0449d71675f5c6077a4ecaf657/udp/server_test.go#L79 about `generic.MulticastOption` ?